### PR TITLE
使用umi ui 做Eslint 检查发现几个问题

### PR DESCRIPTION
**Pull request by leshalv**
修复 https://github.com/ant-design/ant-design-pro/issues/5619

### CODE REVIEW

File: src/components/NoticeIcon/index.tsx [hunk 1]
```diff
@@ -135,6 +133,8 @@ export default class NoticeIcon extends Component<NoticeIconProps> {
     }
   };
 
+  public static Tab: typeof NoticeList = NoticeList;
```
Comment (afc163) [inline src/components/NoticeIcon/index.tsx]: Codacy Issue found: [Declaration of public static field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4526098)

**Comment by leshalv** [comment]
image
@afc163

File: src/pages/user/login/components/Login/index.tsx [hunk 2]
```diff
@@ -122,19 +110,31 @@ class Login extends Component<LoginProps, LoginState> {
...
 
+  public static Tab = LoginTab;
+
+  public static Submit = LoginSubmit;
+
+  public static UserName: React.FunctionComponent<LoginItemProps>;
+
+  public static Password: React.FunctionComponent<LoginItemProps>;
+
+  public static Mobile: React.FunctionComponent<LoginItemProps>;
```
Comment (afc163) [inline src/pages/user/login/components/Login/index.tsx]: Codacy Issue found: [Declaration of public static field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4526098)

File: src/components/NoticeIcon/index.tsx (hunk 1 above)
Comment (afc163) [inline src/components/NoticeIcon/index.tsx]: Codacy Issue found: [Declaration of public static field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4526098)

File: src/pages/user/login/components/Login/index.tsx [hunk 3]
```diff
@@ -122,19 +110,31 @@ class Login extends Component<LoginProps, LoginState> {
     }
   };
 
+  public static Tab = LoginTab;
+
+  public static Submit = LoginSubmit;
+
+  public static UserName: React.FunctionComponent<LoginItemProps>;
+
+  public static Password: React.FunctionComponent<LoginItemProps>;
```
Comment (afc163) [inline src/pages/user/login/components/Login/index.tsx]: Codacy Issue found: [Declaration of public static field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4526098)

File: src/pages/user/login/components/Login/index.tsx [hunk 4]
```diff
@@ -122,19 +110,31 @@ class Login extends Component<LoginProps, LoginState> {
...
+
+  public static Mobile: React.FunctionComponent<LoginItemProps>;
+
+  public static Captcha: React.FunctionComponent<LoginItemProps>;
```
Comment (afc163) [inline src/pages/user/login/components/Login/index.tsx]: Codacy Issue found: [Declaration of public static field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4526098)

File: src/pages/user/login/components/Login/index.tsx [hunk 5]
```diff
@@ -122,19 +110,31 @@ class Login extends Component<LoginProps, LoginState> {
     }
   };
 
+  public static Tab = LoginTab;
+
+  public static Submit = LoginSubmit;
```
Comment (afc163) [inline src/pages/user/login/components/Login/index.tsx]: Codacy Issue found: [Declaration of public static field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4526098)

File: src/pages/user/login/components/Login/index.tsx [hunk 6]
```diff
@@ -122,19 +110,31 @@ class Login extends Component<LoginProps, LoginState> {
     }
   };
 
+  public static Tab = LoginTab;
```
Comment (afc163) [inline src/pages/user/login/components/Login/index.tsx]: Codacy Issue found: [Declaration of public static field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4526098)

File: src/pages/user/login/components/Login/index.tsx [hunk 7]
```diff
@@ -122,19 +110,31 @@ class Login extends Component<LoginProps, LoginState> {
...
+
+  public static Submit = LoginSubmit;
+
+  public static UserName: React.FunctionComponent<LoginItemProps>;
```
Comment (afc163) [inline src/pages/user/login/components/Login/index.tsx]: Codacy Issue found: [Declaration of public static field not allowed after declaration of public instance method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4526098)

**Comment by chenshuai2144** [comment]
这个规则非常的繁琐并且讨人厌，我们已经决定关闭他了。它并不能提升哪怕一点代码质量

**Pull request closed without merging**